dask/array/rechunk.py

```diff
@@ -662,3 +680,86 @@ def format_plan(plan):
     [(3*[10], 2*[15]), ([30], 3*[10])]
     """
     return [format_chunks(c) for c in plan]
+
+
+def _get_chunks(n, chunksize):
+    leftover = n % chunksize
+    n_chunks = n // chunksize
+
+    chunks = [chunksize] * n_chunks
+    if leftover:
+        chunks.append(leftover)
+    return tuple(chunks)
+
+
+def _even_chunksize(N: int, n_chunks: Union[int, float]) -> int:
+    """
+    Find a chunk size that splits an array into even sized chunks.
+
+    Parameters
+    ----------
+    N : int
+        Length of the array
+    n_chunks : int, float.
+        Approximate number of chunks for the array
+
+    Returns
+    -------
+    chunksize : int
+        The chunksize for the array. When an array is chunked with this
+        chunksize,
+
+        * The length of every chunk will be approximately the same.
+        * The number of chunks will be approximately ``n_chunks`` (within one).
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> import dask.array as da
+    >>> x = np.arange(5030)
+    >>> chunksize = da.even_chunksize(len(x), 16)
+    >>> y = da.from_array(x, chunks=chunksize)
+    >>> y.chunks
+    ((296, 296, 296, 296, 296, 296, 296, 296, 296, 296, 296, 296, 296, 296, 296, 296, 294),)
+    >>> len(y.chunks[0])
+    17
```

Comment:
I haven't gone through this logic yet, but a lot of it feels like logic that is already in `rechunk`.  I wonder if you could get away with calling something like the following:

```python
In [5]: x.rechunk(x.shape[0] // 16).chunks                                                                                                                                                                        
Out[5]: 
((314,
  314,
  314,
  314,
  314,
  314,
  314,
  314,
  314,
  314,
  314,
  314,
  314,
  314,
  314,
  314,
  6),)
```

I guess you don't like the last 6 there though?

Reply:
> I guess you don't like the last 6 there though?

Exactly. For Hyperband, each `partial_fit` call is assumed to receive the same amount of data, and `partial_fit` is called on each chunk. In this case, that means  scores after processing 6 and 314 examples are treated the same.